include/sir/ansimacros.h

```diff
@@ -68,68 +82,278 @@
  * 3=italic, 4=underlined, 5=blinking, 7=inverted, 9=strikethrough`).
  * Ends by resetting to the default fg/bg color, and normal attr.
  */
-# define COLOR(attr, fg, bg, s) \
-    SIR_ESC_SEQ(#attr ";" #fg ";" #bg, s) SIR_ESC_SEQE("0;39;49")
+# undef COLOR
+# if !defined(SIR_NO_TEXT_STYLING)
+#  define COLOR(attr, fg, bg, s) \
+     SIR_ESC_SEQ(#attr ";" #fg ";" #bg, s) SIR_ESC_SEQE("0;39;49")
+# else
+#  define COLOR(attr, fg, bg, s) s
```

Comment:
Are we going to get unreferenced variable for the other args? I forget if macros do that or not. (do we need SIR_UNUSED(attr), etc)?

Reply:
No, didn't get any warnings, it's OK with that for macros.